x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -746,7 +749,14 @@ public static void testFunctionInfo() {
             );
         }
 
-        Set<String> returnTypes = Arrays.stream(description.returnType()).collect(Collectors.toCollection(TreeSet::new));
+        Set<String> underConstructionTypes = DataType.UNDER_CONSTRUCTION.keySet()
```

Comment:
I think I just fixed this same issue in a different way: https://github.com/elastic/elasticsearch/pull/114056/files#diff-5b3a7bcaf3f57ee42ca359ddd440590dae3f4ddb24071a7587659ef6ad941fe3 (in main; backport pending.)

Reply:
Your approach is nicer - so I reverted these changes